controllers/costmanagement_controller.go

```diff
@@ -147,13 +150,28 @@ func ReflectSpec(r *CostManagementReconciler, cost *costmgmtv1alpha1.CostManagem
 		costConfig.UploadCycle = costmgmtv1alpha1.DefaultUploadCycle
 	}
 
-	costConfig.SourceName = StringReflectSpec(r, cost, &cost.Spec.Source.SourceName, &cost.Status.Source.SourceName, "")
+	var sourceNameChanged bool
+	costConfig.SourcesAPIPath, _ = StringReflectSpec(r, cost, &cost.Spec.Source.SourcesAPIPath, &cost.Status.Source.SourcesAPIPath, costmgmtv1alpha1.DefaultSourcesPath)
+	costConfig.SourceName, sourceNameChanged = StringReflectSpec(r, cost, &cost.Spec.Source.SourceName, &cost.Status.Source.SourceName, "")
 	costConfig.CreateSource = false
 	if cost.Spec.Source.CreateSource != nil {
 		costConfig.CreateSource = *cost.Spec.Source.CreateSource
 	}
+	sourceCycleChange := false
+	if !reflect.DeepEqual(cost.Spec.Source.CheckCycle, cost.Status.Source.CheckCycle) {
+		cost.Status.Source.CheckCycle = cost.Spec.Source.CheckCycle
+		sourceCycleChange = true
+	}
+	if cost.Status.Source.CheckCycle != nil {
+		costConfig.SourceCheckCycle = *cost.Status.Source.CheckCycle
+	} else {
+		costConfig.SourceCheckCycle = costmgmtv1alpha1.DefaultSourceCheckCycle
+	}
+	if sourceNameChanged || sourceCycleChange {
+		costConfig.LastSourceCheckTime = cost.Status.Source.LastSourceCheckTime
+	}
```

Comment:
yeah the intention was the flip of this 🤦  if these items were false. My screw up with the last minute change to better handle CR updates. Will fix this.

Reply:
updated.

Reply:
Things are looking better, but we need another check somewhere. I tried creating a source, but I failed authentication. The source was not created, but because of the above change, the time _was_ set, and `CheckSource` return `false`. Maybe we need additional logic in `CheckSource` that looks at spec `create_source` and status `source_defined` and returns `true` if `create=true` and `defined=false`?

Reply:
What happens if you delete the `source` section from the spec save it then add it back. That should force the name to go from a value, to nil, to a value and trigger the source check.

If you get a 401 or 403 I don't think we would want it to keep trying just because `create=true".  Maybe that is not a great experience 🤷 ?

Reply:
hmm, that's a good point. I'll give that a try to see what happens.

And yes, thinking about it now, if we're hitting auth errors, it would not be good to watch this loop over and over trying to create something that will never be successful without intervention.